applications/RomApplication/python_scripts/rom_manager.py

```diff
@@ -256,262 +227,264 @@ def RunHROM(self, mu_run=[None], use_full_model_part = False):
         else:
             err_msg = f'Provided projection strategy {chosen_projection_strategy} is not supported. Available options are \'galerkin\', \'lspg\' and \'petrov_galerkin\'.'
             raise Exception(err_msg)
-        self.__LaunchRunHROM(mu_run, use_full_model_part)
+        self._LaunchRunHROM(mu_run, use_full_model_part)
+
+
+
+    def ComputeErrors(self, mu_list, case="Fit"):
+        fom_snapshots = self.data_base.get_snapshots_matrix_from_database(mu_list, table_name=f'FOM')
+        if case=="Fit":
+            stages = self.general_rom_manager_parameters["rom_stages_to_train"].GetStringArray()
+        elif case=="Test":
+            stages = self.general_rom_manager_parameters["rom_stages_to_test"].GetStringArray()
+        stages = {"ROM", "HROM"} & set(stages)  # Ensures only "ROM" or "HROM" if present
+        rom_snapshots = None
+        hrom_snapshots = None
+        if "ROM" in stages:
+            rom_snapshots = self.data_base.get_snapshots_matrix_from_database(mu_list, table_name=f'ROM')
+            error_rom_fom = np.linalg.norm(fom_snapshots - rom_snapshots) / np.linalg.norm(fom_snapshots)
+            setattr(self, f'ROMvsFOM_{case}', error_rom_fom)
+        if "HROM" in stages:
+            if rom_snapshots is None:  # Only fetch if not already fetched
+                rom_snapshots = self.data_base.get_snapshots_matrix_from_database(mu_list, table_name=f'ROM')
+            hrom_snapshots = self.data_base.get_snapshots_matrix_from_database(mu_list, table_name=f'HROM')
+            error_rom_hrom = np.linalg.norm(rom_snapshots - hrom_snapshots) / np.linalg.norm(rom_snapshots)
+            error_fom_hrom = np.linalg.norm(fom_snapshots - hrom_snapshots) / np.linalg.norm(fom_snapshots)
+            setattr(self, f'ROMvsHROM_{case}', error_rom_hrom)
+            setattr(self, f'FOMvsHROM_{case}', error_fom_hrom)
```

Comment:
Not super fan of using this, why don't you use a dictionary instead?

```python
self.ROMvsHROM['Fit']
```

for example?

Reply:
now using dictionaries